### PR TITLE
🦄️ refactor(DatePicker,Calendar): dateRender,monthRender => cellRender

### LINKED ISSUE #41255: DatePicker 组件在 picker=month｜quarter｜year 时，  dateRender这个函数不生效
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/ji-ben-antd-5-3-1-forked-qqptm5?file=/demo.tsx:486-496)

### Steps to reproduce

打开demo即可看到 

### What is expected?

期望能函数能够正常运行

### What is actually happening?

在 picker=month｜quarter｜year 时，并不生效

| Environment | Info |
| --- | --- |
| antd | 5.3.1 |
| React | React17/18 |
| System | Mac |
| Browser | Chrome 版本 111.0.5563.64（正式版本） (arm64) |

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [X] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #41255
- #41255
- https://github.com/ant-design/ant-design/discussions/41303

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  dateRender,monthRender => cellRender |
| 🇨🇳 Chinese |  dateRender,monthRender => cellRender  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 9b0ad64</samp>

Added a new `cellRender` prop to the `DatePicker` and `RangePicker` components to enable custom rendering of any picker cells. Updated the documentation, demos, and dependency to reflect this change. Removed the deprecated `dateRender` and `monthCellRender` props.

### <samp>🤖 Generated by Copilot at 9b0ad64</samp>

*  Rename `dateRender` prop to `cellRender` and add support for different types of picker cells in `DatePicker` and `RangePicker` components ([link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2L4-R4), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2R241-R252), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2R293), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-f07b7c62bb0eff9e7cedd676d4358cd78bac87230155102d79a2af2315bc17b9L9-R9), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-f07b7c62bb0eff9e7cedd676d4358cd78bac87230155102d79a2af2315bc17b9L23-R23), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-278af64bde59169ca85bf2d7df8a110094a7c153d88b5277c1614c365e65b910L84-R84), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-278af64bde59169ca85bf2d7df8a110094a7c153d88b5277c1614c365e65b910L167), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-4f6ca776b129678ab875461618596d96f83d09613799c0180d0489e7486f0cdeL85-R85), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-4f6ca776b129678ab875461618596d96f83d09613799c0180d0489e7486f0cdeL168), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-4f6ca776b129678ab875461618596d96f83d09613799c0180d0489e7486f0cdeL187-R188), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L138-R138))
*  Update `rc-picker` dependency to `~3.5.0` to use the `CellRenderInfo` type and the `cellRender` prop ([link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L138-R138))
*  Change demo code file name from `date-render.tsx` to `cell-render.tsx` and update import order of `React` and `antd` ([link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-f07b7c62bb0eff9e7cedd676d4358cd78bac87230155102d79a2af2315bc17b9L1-R2), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-278af64bde59169ca85bf2d7df8a110094a7c153d88b5277c1614c365e65b910L30-R30), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-4f6ca776b129678ab875461618596d96f83d09613799c0180d0489e7486f0cdeL31-R31), [link](https://github.com/ant-design/ant-design/pull/41584/files?diff=unified&w=0#diff-aca4c6bd851202f1483f32f7a996814a1be3235dd4535260b28a1bc149fa7378))

### CODE REVIEW

File: components/date-picker/index.zh-CN.md [hunk 1]
```diff
@@ -186,7 +185,7 @@ import locale from 'antd/locale/zh_CN';
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
 | allowEmpty | 允许起始项部分为空 | \[boolean, boolean] | \[false, false] |  |
-| dateRender | 自定义日期单元格的内容。`info` 参数自 4.3.0 添加 | function(currentDate: dayjs, today: dayjs, info: { range: `start` \| `end` }) => React.ReactNode | - |  |
+| cellRender | 自定义单元格的内容。 | function(current: dayjs, today: dayjs, info: { originNode: React.ReactElement,today: DateType, range?: 'start' \| 'end', type: PanelMode, locale?: Locale, subType?: 'hour' \| 'minute' \| 'second' \| 'meridiem' }) => React.ReactNode | - | 5.4.0 |
```
Comment (afc163) [inline components/date-picker/index.zh-CN.md:188]: 找个地方写一下 antd 版本低于 5.4.0 时用哪些 API。

File: components/calendar/generateCalendar.tsx [hunk 2]
```diff
@@ -278,8 +288,7 @@ function generateCalendar<DateType>(generateConfig: GenerateConfig<DateType>) {
           prefixCls={prefixCls}
           locale={contextLocale?.lang}
           generateConfig={generateConfig}
-          dateRender={dateRender}
-          monthCellRender={(date) => monthRender(date, contextLocale?.lang)}
```
Comment (afc163) [inline components/calendar/generateCalendar.tsx:282]: 用户使用旧的 API 要给 warning 信息，类型上给废弃标。

Comment (kiner-tang) [reply components/calendar/generateCalendar.tsx:282]: done

File: components/date-picker/index.zh-CN.md (hunk 1 above)
Comment (kiner-tang) [reply components/date-picker/index.zh-CN.md:190]: done

**Review by afc163 — APPROVED** [review]

**Pull request merged**